### PR TITLE
PLAT-5461: Print all API requests and result headers

### PR DESCRIPTION
**Pull request by gadyaari**

### CODE REVIEW

File: lib/BackendClient.js [hunk 1]
```diff
@@ -201,35 +201,35 @@ var BackendClient = (function(){
         return deferred.promise;
     }
 
-    function registerEntryInDatabase(entryId, serverIndex, state, event, logger) {
+    function registerEntryInDatabase(entryId, serverIndex, state, event) {
         var deferred = Q.defer();
-        logger.debug("Request for register media server started");
+        logger.debug("[%s] Calling registerMediaServer. Hostname: %s, serverIndex: %s, Application: %s, state: %s", entryId, mediaServerHostname, serverIndex, applicationName, state);
         client.liveStream.registerMediaServer(function(result, err, headers) {
+            printAPIResponse(logger, result, headers);
```
Comment (yossipapi) [inline lib/BackendClient.js]: Use class logger since you removed the function param called logger

File: lib/BackendClient.js [hunk 2]
```diff
@@ -33,10 +33,10 @@ var BackendClient = (function(){
     var BackendClient = {};
     var mediaServerIdPromise = null;
 
-    function printAPIResponse(logger, results, headers) {
+    function printAPIResponse(logger, result, headers) {
         logger.info("API call headers: %j", headers);
-        if(results)
-            logger.info("API call results: %j", results);
+        if(result)
+            logger.info("API call results: %j", result);
```
Comment (yossipapi) [inline lib/BackendClient.js]: Use API Call Result and not Results

File: lib/BackendClient.js [hunk 3]
```diff
@@ -185,7 +185,7 @@ var BackendClient = (function(){
         client.doMultiRequest(function(results, err, headers) {
             var endTime = new Date();
             var operationTime = endTime - startTime;
-            logger.info("Got server node info for entry %s (took %d)", entryId, operationTime);
+            logger.info("[%s] Got server node info for entry (took %d)", entryId, operationTime);
```
Comment (yossipapi) [inline lib/BackendClient.js]: Remove the string "for entry"

File: lib/BackendClient.js [hunk 4]
```diff
@@ -33,10 +33,10 @@ var BackendClient = (function(){
     var BackendClient = {};
     var mediaServerIdPromise = null;
 
-    function printAPIResponse(logger, results, headers) {
```
Comment (yossipapi) [inline lib/BackendClient.js]: Remove the logger attribute from the function and use class logger

File: lib/BackendClient.js [hunk 5]
```diff
@@ -251,19 +251,17 @@ var BackendClient = (function(){
             });
     };
 
-    BackendClient.registerEntryInDatabase = function(entryObject, state, event, logger) {
-        logger.info("Calling registerMediaServer, updating entry in database");
+    BackendClient.registerEntryInDatabase = function(entryObject, state, event) {
```
Comment (yossipapi) [inline lib/BackendClient.js]: Print log if create session failed

File: lib/BackendClient.js (hunk 4 above)
Comment (gadyaari) [inline lib/BackendClient.js]: done

File: lib/BackendClient.js (hunk 2 above)
Comment (gadyaari) [inline lib/BackendClient.js]: done

File: lib/BackendClient.js (hunk 3 above)
Comment (gadyaari) [inline lib/BackendClient.js]: done

File: lib/BackendClient.js (hunk 1 above)
Comment (gadyaari) [inline lib/BackendClient.js]: done

File: lib/BackendClient.js (hunk 5 above)
Comment (gadyaari) [inline lib/BackendClient.js]: done

**Pull request merged**